examples/cms/src/FieldMapping.tsx

```diff
@@ -0,0 +1,235 @@
+}
+
+const FieldMappingRow = memo(({ field, disabled, onDisable, onNameChange }: FieldMappingRowProps) => {
+    return (
+        <>
+            <button
+                type="button"
+                className="source-field"
+                aria-disabled={disabled}
+                onClick={() => onDisable(field.id)}
+                tabIndex={0}
+            >
+                <input type="checkbox" checked={!disabled} tabIndex={-1} />
+                <span>{field.id}</span>
+            </button>
+            <ChevronIcon />
+            <input
+                type="text"
+                style={{
+                    width: "100%",
+                    opacity: disabled ? 0.5 : 1,
+                }}
+                disabled={disabled}
+                placeholder={field.id}
+                value={field.name}
+                onChange={event => onNameChange(field.id, event.target.value)}
+                onKeyDown={event => {
+                    if (event.key === "Enter") {
+                        event.preventDefault()
+                    }
+                }}
+            />
+        </>
+    )
+})
+
+const initialManagedCollectionFields: ManagedCollectionFields = []
+const initialFieldIds: FieldIds = new Set()
+
+interface FieldMappingProps {
+    collection: ManagedCollection
+    dataSource: DataSource
+    initialSlugFieldId: string | null
+}
+
+export function FieldMapping({ collection, dataSource, initialSlugFieldId }: FieldMappingProps) {
+    const [status, setStatus] = useState<"mapping-fields" | "loading-fields" | "syncing-collection">("loading-fields")
+    const isSyncing = status === "syncing-collection"
+    const isLoadingFields = status === "loading-fields"
+
+    const sourceFields = useMemo(() => computeFieldsFromDataSource(dataSource), [dataSource])
+    const possibleSlugFields = useMemo(() => sourceFields.filter(field => field.type === "string"), [sourceFields])
+
+    const [selectedSlugField, setSelectedSlugField] = useState<ManagedCollectionField | null>(
+        possibleSlugFields.find(field => field.id === initialSlugFieldId) ?? possibleSlugFields[0] ?? null
+    )
+
+    const [fields, setFields] = useState(initialManagedCollectionFields)
+    const [ignoredFieldIds, setIgnoredFieldIds] = useState(initialFieldIds)
+
+    useEffect(() => {
+        const abortController = new AbortController()
+
+        collection
+            .getFields()
+            .then(collectionFields => {
+                if (abortController.signal.aborted) return
+
+                setFields(mergeFieldsWithExistingFields(sourceFields, collectionFields))
+
+                const existingFieldIds = new Set(collectionFields.map(field => field.id))
+                const ignoredFields = sourceFields.filter(sourceField => !existingFieldIds.has(sourceField.id))
+
+                if (initialSlugFieldId) {
+                    setIgnoredFieldIds(new Set(ignoredFields.map(field => field.id)))
+                }
```

Comment:
I guess this if condition might be hard to understand. Can we just check if the existing field ids has a size of zero? Would that be easier to understand? Either way we should probably add a comment

Reply:
We can't, because when syncing for the first time `existingFields` would be empty.

Also this iteration allows to sync even if there is no field selected, if there is a valid slug selected then syncing will work!